src/aioauth/server.py

```diff
@@ -22,17 +30,75 @@
     TokenInactiveIntrospectionResponse,
 )
 from .structures import CaseInsensitiveDict
-from .types import ResponseType
-from .utils import build_uri, catch_errors_and_unavailability, decode_auth_headers
+from .types import GrantType, RequestMethod, ResponseType
+from .utils import (
+    build_uri,
+    catch_errors_and_unavailability,
+    decode_auth_headers,
+    is_secure_transport,
+    str_to_list,
+)
 
+AnyResponseTypeClass = Union[
+    Type[ResponseTypeBase],
+    Type[ResponseTypeToken],
+    Type[ResponseTypeAuthorizationCode],
+    Type[ResponseTypeNone],
+]
+
+AnyGrantTypeClass = Union[
+    Type[AuthorizationCodeGrantType],
+    Type[ClientCredentialsGrantType],
+    Type[PasswordGrantType],
+    Type[RefreshTokenGrantType],
+    Type[GrantTypeBase],
+]
+
+
+class AuthorizationServer:
+    def __init__(
+        self,
+        db: BaseDB,
+        response_types: Optional[Dict] = None,
+        grant_types: Optional[Dict] = None,
+    ):
+        if response_types is None:
+            self.response_types = {
+                ResponseType.TYPE_TOKEN: ResponseTypeToken,
+                ResponseType.TYPE_CODE: ResponseTypeAuthorizationCode,
+                ResponseType.TYPE_NONE: ResponseTypeNone,
+            }
+        else:
+            self.response_types = response_types
+
+        if grant_types is None:
+            self.grant_types = {
+                GrantType.TYPE_AUTHORIZATION_CODE: AuthorizationCodeGrantType,
+                GrantType.TYPE_CLIENT_CREDENTIALS: ClientCredentialsGrantType,
+                GrantType.TYPE_PASSWORD: PasswordGrantType,
+                GrantType.TYPE_REFRESH_TOKEN: RefreshTokenGrantType,
+            }
+        else:
+            self.grant_types = grant_types
+        self.db = db
```

Comment:
Why not default the values via the `__init__`, as opposed to using `None` for `response_types` and `grant_types`?

Reply:
I did so because of the mutable arguments in python a bit dangerous: https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/#:~:text=In%20Python%2C%20when%20passing%20a,or%20even%20a%20class%20instance.